source/perflib/basename.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+
+/*
+ * windows variant of basename(3). works on ASCIIZ only.
+ * unlike unix there two variants of path separators on
+ * on windows: slash and backslash.
+ */
+const char *basename(char *path)
+{
+    char *slash, *bslash, *rv;
+
+    if (path == NULL || *path == '\0')
+        return ".";
+
+    slash = strrchr(path, '/');
+    bslash = strrchr(path, '\\');
+    rv = (slash > bslash) ? slash : bslash;
```

Comment:
Please don't use `>` to compare pointers that might be `NULL` - while this often works, [it's technically undefined behaviour](https://stackoverflow.com/a/35252004/7761803)